src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -417,7 +417,7 @@ namespace cryptonote
           storage_ok = false;
         }
 
-        if (!epee::net_utils::is_ip_public(m_sn_public_ip)) {
+        if (!tools::dns_utils::is_ip_public(tools::dns_utils::swap_byte_order(m_sn_public_ip))) {
```

Comment:
An unconditional swap here doesn't seem right.

I think, for backwards compatibility with epee's perversity, we might have to do a swap on little-endian but not a big-endian?  Or maybe I have that wrong.

If we have some IP such as 1.2.3.4, how is that *currently* (before this PR) getting encoded on the wire?  Either:
"\x01\x02\x03\x04" (which is big-endian) or "\x04\x03\x02\x01" (little endian).  I *think* the latter is what's happening?

What we *want* is the uint32_t value 0x01020304, so I'm wondering what we actually get inside `m_sn_public_ip` currently -- is it 0x01020304 or 0x04030201?

Reply:
I believe `m_sn_public_ip` is big endian bytes and is the output of a call to `in_addr_t inet_addr(const char *cp);` But the computer is reading them as a little endian uint. swap_byte_order puts them back into little endian for `is_ip_public()`. Unsure what they actually get sent over the wire as atm will do some more research

Reply:
Maybe something like:
```C++
struct ip_address {
    std::array<uint8_t, 4> octets;
    uint32_t as_host32() const {
        return (octets[0] << 24) | (octets[1] << 16) | (octets[2] << 8) | octets[3];
    }
}
```